src/lib/socks/headers.ts

```diff
@@ -484,3 +560,20 @@ export function interpretResponseBuffer(buffer:ArrayBuffer) : Response {
 
   return response;
 }
+
+// Helper methods for encoding strings to/from ArrayBuffer
+
+// ArrayBuffer to string
+function ab2str(buf:ArrayBuffer) : string {
```

Comment:
We can use `Buffer` rather than (re-)adding these helper functions - I gave examples above where you've used these.